src/Api/Suppression/Whitelist.php

```diff
@@ -0,0 +1,149 @@
+     *
+     * @return IndexResponse
+     */
+    public function index(string $domain, int $limit = 100)
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::range($limit, 1, 10000, 'Limit parameter must be between 1 and 10000');
+
+        $params = [
+            'limit' => $limit,
+        ];
+
+        $response = $this->httpGet(sprintf('/v3/%s/whitelists', $domain), $params);
+
+        return $this->hydrateResponse($response, IndexResponse::class);
+    }
+
+    /**
+     * @param string $domain  Domain to show whitelist for
+     * @param string $address whitelist address
+     *
+     * @return ShowResponse
+     */
+    public function show(string $domain, string $address)
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::stringNotEmpty($address);
+
+        $response = $this->httpGet(sprintf('/v3/%s/whitelists/%s', $domain, $address));
+
+        return $this->hydrateResponse($response, ShowResponse::class);
+    }
+
+    /**
+     * @param string $domain  Domain to create whitelist for
+     * @param string $address whitelist email address or domain name
+     * @param array  $params  optional
+     *
+     * @return CreateResponse
+     */
+    public function create(string $domain, string $address, array $params = [])
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::stringNotEmpty($address);
+
+        if (false !== filter_var($address, FILTER_VALIDATE_EMAIL)) {
+            $params['address'] = $address;
+        } elseif (false !== filter_var($address, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME)) {
+            $params['domain'] = $address;
+        } else {
+            throw new InvalidArgumentException('Address should be valid email or domain name');
+        }
+
+        $response = $this->httpPost(sprintf('/v3/%s/whitelists', $domain), $params);
+
+        return $this->hydrateResponse($response, CreateResponse::class);
+    }
+
+    /**
+     * @param string $domain   Domain to create whitelist for
+     * @param string $filePath csv file path
+     * @param array  $params   optional
+     *
+     * @return ImportResponse
+     */
+    public function import(string $domain, string $filePath, array $params = [])
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::stringNotEmpty($filePath);
+        Assert::fileExists($filePath);
+
+        $response = $this->httpPost(
+            sprintf('/v3/%s/whitelists/import', $domain),
+            ['file' => fopen($filePath, 'r')],
+            array_merge($params, [
+                'filename' => basename($filePath),
+            ])
+        );
+
+        return $this->hydrateResponse($response, ImportResponse::class);
+    }
+
+    /**
+     * @param string $domain  Domain to delete whitelist for
+     * @param string $address whitelist address
+     * @param array  $params  optional
+     *
+     * @return DeleteResponse
+     */
+    public function delete(string $domain, string $address, array $params = [])
```

Comment:
The same point for the `$params` variable, as mentioned previously

https://documentation.mailgun.com/en/latest/api-suppressions.html#delete-a-single-record-from-whitelist-table

Reply:
removed